### ISSUE TITLE
Remove OpenSSL dependency

### ISSUE THREAD

**Issue by nrkramer** [ZigRazor/CXXGraph#274]
Please remove the OpenSSL dependency. Tying a dependency as well-known as OpenSSL for networking capabilities (and subsequently needing to ship it with compiled binaries) may set off alarms for users.

If a robust sha algorithm is needed, [Hash](https://github.com/Chocobo1/Hash) is a possible alternative.

**Comment by rockingdice** [comment]
No offense, but why should a graph library use OpenSSL? Why involve a network library? I thought it was a purely mathematical problem solver, I'm a bit confused because I found this repo from Google and don't know the details of this library.
Actually, I'm looking for a library that could provide an easy-to-use solution to solve graph problems, I found this library is promising but this issue confuses me.

**Comment by nrkramer** [comment]
Hey @rockingdice, that's the reason I raised this issue.

I believe OpenSSL is currently being used for it's excellent hashing capabilities, which is necessary for making some graph implementations very fast.

@ZigRazor has mentioned it is slated for removal, which should clear this issue up. I've looked at the source code, and no SSL sockets are being opened or used, so the dependency isn't being used in a malicious way - although it makes developing against the library a bit more cumbersome.

Either way, speedy removal seems essential for mass-adoption of this library, as it has incredible promise imo.
